Source/ASRunLoopQueue.mm

```diff
@@ -469,3 +489,230 @@ - (void)unlock
 }
 
 @end
+
+#pragma mark - ASCATransactionQueue
+
+@interface ASCATransactionQueue () {
+  CFRunLoopRef _runLoop;
+  CFRunLoopSourceRef _runLoopSource;
+  CFRunLoopObserverRef _runLoopObserver;
+  NSPointerArray *_internalQueue; // Use NSPointerArray so we can decide __strong or __weak per-instance.
+  ASDN::RecursiveMutex _internalQueueLock;
+  BOOL _disableInterfaceStateCoalesce;
+
+  // In order to not pollute the top-level activities, each queue has 1 root activity.
+  os_activity_t _rootActivity;
+
+#if ASRunLoopQueueLoggingEnabled
+  NSTimer *_runloopQueueLoggingTimer;
+#endif
```

Comment:
Can we not move the duplicated ivars up into ASAbstractRunLoopQueue?

Reply:
The comment on `_internalQueue` doesn't apply to this class; we always use it in strong mode.